lib/resources/context.d.ts

```diff
@@ -129,31 +121,38 @@ export interface StandardBindingContextTransforms {
 	checkedValue: BindingContextIdentityTransform<any>
 	options: BindingContextIdentityTransform<any>
 	optionsText: BindingContextIdentityTransform<string>
+	optionsCaption: BindingContextIdentityTransform<string>
+	optionsValue: BindingContextIdentityTransform<string>
 	selectedOptions: BindingContextIdentityTransform<any>
 	uniqueName: BindingContextIdentityTransform<boolean>
 	template: BindingContextIdentityTransform<any>
 	component: BindingContextIdentityTransform<string | { name: any, params: any }>
 	if: BindingContextIdentityTransform<unknown>
 	ifnot: BindingContextIdentityTransform<boolean>
 
-	// TODO: Use this code instead when const string types are supported by typescript!
-	// The generic type As should NEVER be used. It exists to keep the const string type. (https://github.com/microsoft/TypeScript/issues/30680)
-	// foreach: <Value extends MaybeReadonlyObservableArray<Data> | MaybeReadonlyObservable<{ data: Data, as: As }>, Data, Context extends BindingContext, As extends string>(value: Value, parentContext: Context) =>
-	// 	Value extends { data: MaybeObservableArray<infer T>, as: string } ?
-	// 		Overlay<Record<Value['as'], T>, Context> :
-	// 		Overlay<ChildBindingContext<Data, Context>, Context>
-
-	// Since we do not have support for const string types in Typescript yet (see https://github.com/microsoft/TypeScript/issues/30680), we resort to simpler matching here and instead
-	// we do more comprehensive analysis for 'foreach' bindings in-code instead.
-	foreach: <Value extends MaybeReadonlyObservableArray<any> | MaybeReadonlyObservable<any>, Context extends BindingContext<any, {}>>(value: Value, parentContext: Context) =>
-		Value extends Record<'as', infer T> & Record<'data', MaybeReadonlyObservableArray<(infer V)>> ?
-			T extends string ? BindingOverlay<Record<T, V>, Overlay<ChildBindingContext<V, Context>, Context>> :
+	// Since we do not have support for const string types in Typescript yet (see https://github.com/microsoft/TypeScript/issues/30680), we resort to a complex type 'Narrow' to make sure
+	// it is not widened to a string. Inspiration from ts-toolbelt.
+	foreach: <Key, VM, Context extends BindingContext>(value: { data: MaybeReadonlyObservableArray<VM>, as: Narrow<Key> } | MaybeReadonlyObservableArray<VM>, parentContext: Context) =>
+		Key extends string ? string extends Key ?
 			'\'as\' must be a string literal' :
```

Comment:
Cannot the generic type `Key` extend `string`. If so the `'\'as\' must be a string literal'` is unnecessary, because it will `never` happen.